src/strategy/strategy.ts

```diff
@@ -10,14 +11,139 @@ export interface StrategyTransportInterface {
 export interface Constraint {
   contextName: string;
   operator: Operator;
+  inverted: boolean;
   values: string[];
+  value?: string | number | Date;
+  caseInsensitive?: boolean;
 }
 
 export enum Operator {
-  IN = <any>'IN',
-  NOT_IN = <any>'NOT_IN',
+  IN = 'IN',
+  NOT_IN = 'NOT_IN',
+  STR_ENDS_WITH = 'STR_ENDS_WITH',
+  STR_STARTS_WITH = 'STR_STARTS_WITH',
+  STR_CONTAINS = 'STR_CONTAINS',
+  NUM_EQ = 'NUM_EQ',
+  NUM_GT = 'NUM_GT',
+  NUM_GTE = 'NUM_GTE',
+  NUM_LT = 'NUM_LT',
+  NUM_LTE = 'NUM_LTE',
+  DATE_AFTER = 'DATE_AFTER',
+  DATE_BEFORE = 'DATE_BEFORE',
+  SEMVER_EQ = 'SEMVER_EQ',
+  SEMVER_GT = 'SEMVER_GT',
+  SEMVER_LT = 'SEMVER_LT',
 }
 
+export type OperatorImpl = (constraint: Constraint, context: Context) => boolean;
+
+const cleanValues = (values: string[]) => values
+  .filter(v => !!v)
+  .map(v => v.trim())
+
+const InOperator = (constraint: Constraint, context: Context) => {
+  const field = constraint.contextName;
+  const values = cleanValues(constraint.values);
+  const contextValue = resolveContextValue(context, field);
+  if(!contextValue) {
+    return false;
+  }
+  
+  const isIn = values.some(val => val === contextValue);
+  return constraint.operator === Operator.IN ? isIn : !isIn;
+}
+
+const StringOperator = (constraint: Constraint, context: Context) => {
+  const { contextName, operator, caseInsensitive} = constraint;
+  let values = cleanValues(constraint.values);
+  let contextValue = resolveContextValue(context, contextName);
+
+  if(caseInsensitive) {
+    values = values.map(v => v.toLocaleLowerCase());
+    contextValue = contextValue?.toLocaleLowerCase();
+  }
+
+  if(operator === Operator.STR_STARTS_WITH) {
+    return values.some(val => contextValue?.startsWith(val)); 
+  } if(operator === Operator.STR_ENDS_WITH) {
+    return values.some(val => contextValue?.endsWith(val));
+  } if(operator === Operator.STR_CONTAINS) {
+    return values.some(val => contextValue?.includes(val));
+  } 
+    return false;
+}
+
+const SemverOperator = (constraint: Constraint, context: Context) => {
+  const { contextName, operator} = constraint;
+  const value = constraint.value as string;
+  const contextValue = resolveContextValue(context, contextName);
+  if(!contextValue) {
+    return false;
+  }
+
+  if(operator === Operator.SEMVER_EQ) {
+    return semverEq(contextValue, value); 
+  } if(operator === Operator.SEMVER_LT) {
+    return semverLt(contextValue, value); 
+  } if(operator === Operator.SEMVER_GT) {
+    return semverGt(contextValue, value); 
+  } 
+    return false;
+}
+
+const DateOperator = (constraint: Constraint, context: Context) => {
+  const { operator } = constraint;
+  const value = new Date(constraint.value as string);
+  const currentTime = context.currentTime ? new Date(context.currentTime) : new Date();
```

Comment:
we might want to move this to the creation of the context. 

+ The benefit will be that the date and time will not change between multiple constriants within the same feature toggle. 
- The downside would be that we would have to create the currentTime date instance on the context, even when a use is not using any constraints that uses date operators.